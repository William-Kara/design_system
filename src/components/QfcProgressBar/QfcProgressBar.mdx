import { Meta, ArgTypes } from '@storybook/blocks';
import QfcProgressBar from '@/components/QfcProgressBar/QfcProgressBar.vue';

<Meta title="Elements/ProgressBar" />

## Description

The QfcProgressBar component is a customizable progress bar used to display the progress of different tasks.

## Props

The following props are available for the QfcProgressBar component:

<ArgTypes of={QfcProgressBar} />

You should add title to QfcProgressBar for better accessibility.
