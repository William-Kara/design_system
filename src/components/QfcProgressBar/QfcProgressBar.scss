
.qfc-progress-bar {
  display: grid;
  grid-template-columns: 1fr 36px;
  align-items: center;
  padding: 5px;
  &__line {
    float: left;
    border-radius: 100px 100px 100px 100px;
    height: 7px;
    background-color: #696969;
    margin-right: 2rem;
  }
    &__completed {
      border-radius: 100px 100px 100px 100px;
      height: 7px;
    }
  
  
  &__value {
  float: right;
  position: relative;
}
}