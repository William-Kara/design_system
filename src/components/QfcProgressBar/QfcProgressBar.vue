<template>
  <div
    class="qfc-progress-bar"
    role="progressbar"
    :title="title"
    :aria-valuenow="value"
    aria-valuemin="0"
    aria-valuemax="100"
  >
    <div class="qfc-progress-bar__line">
      <Transition>
        <div class="qfc-progress-bar__completed" :style="completedStyle"></div>
      </Transition>
    </div>
    <div class="qfc-progress-bar__value">{{ percentage }}%</div>
  </div>
</template>

<script lang="ts" setup>
import { computed } from 'vue';

type Color = `#${string}`;
interface Props {
  value: number;
  title: string;
  color?: Color;
}

const props = defineProps<Props>();

const percentage = computed(() => {
  return props.value <= 100 ? props.value : 100;
});
const completedStyle = computed(() => {
  return { width: percentage.value + '%', 'background-color': props.color };
});
</script>

<style lang="scss" scoped src="./QfcProgressBar.scss" />
