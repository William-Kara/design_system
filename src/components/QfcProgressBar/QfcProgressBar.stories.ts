import QfcProgressBar from './QfcProgressBar.vue';
import type { Meta, StoryObj } from '@storybook/vue3';

const meta: Meta<typeof QfcProgressBar> = {
  component: QfcProgressBar,
  title: 'Elements/ProgressBar',
  argTypes: {},
};
export default meta;

type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    color: '#F5DE8D',
    value: 10,
    title: 'Lesson Progression',
  },
};

export const ProgressBar50: Story = {
  args: {
    color: '#ffc0cb',
    value: 50,
    title: 'Lesson Progression',
  },
};

export const ProgressBar100: Story = {
  args: {
    color: '#44A146',
    value: 100,
    title: 'Lesson Progression',
  },
};
