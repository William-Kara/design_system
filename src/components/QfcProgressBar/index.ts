import QfcProgressBar from './QfcProgressBar.vue';

export { QfcProgressBar };
export default QfcProgressBar;
