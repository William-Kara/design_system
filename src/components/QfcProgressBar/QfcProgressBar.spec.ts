import { it, describe, beforeEach, expect } from 'vitest';

import { shallowMount } from '@vue/test-utils';
import QfcProgressBar from './QfcProgressBar.vue';

let wrapper: any = null;

const propsValues = {
  color: '#F5DE8D',
  value: 10,
  title: 'Lesson Progression',
};

beforeEach(() => {
  wrapper = shallowMount(QfcProgressBar, {
    propsData: propsValues,
  });
});

describe('QfcProgressBar Component', () => {
  it('should mount properly', () => {
    expect(wrapper).toBeTruthy();
  });

  it('should have correct props', () => {
    expect(wrapper.props()).toEqual(propsValues);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
