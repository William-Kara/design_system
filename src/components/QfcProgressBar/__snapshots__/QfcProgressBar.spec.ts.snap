// Vitest Snapshot v1

exports[`QfcProgressBar Component > should render correctly 1`] = `
"<div data-v-2abc2a22=\\"\\" class=\\"qfc-progress-bar\\" role=\\"progressbar\\" title=\\"Lesson Progression\\" aria-valuenow=\\"10\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
  <div data-v-2abc2a22=\\"\\" class=\\"qfc-progress-bar__line\\">
    <transition-stub data-v-2abc2a22=\\"\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
      <div data-v-2abc2a22=\\"\\" class=\\"qfc-progress-bar__completed\\" style=\\"width: 10%; background-color: rgb(245, 222, 141);\\"></div>
    </transition-stub>
  </div>
  <div data-v-2abc2a22=\\"\\" class=\\"qfc-progress-bar__value\\">10%</div>
</div>"
`;
